### PR TITLE
LPS-182336 Adaptive Media's Adapt Remaining feature with S3 store seems to be leaking HTTP connections

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
During the "Adapt Remaining" process, `InputStream`s are not closed after usage leading to a congestion of connection resources when using a remote repository like S3. This is caused by the `ImageReader` object whose `dispose()` method does not close the `InputStream` handed to it.

Proposed Solution
========
Adding logic in the `finally` clause that closes the `InputStream`. Please note that closing the `imageInputStream` object won't automatically close the `inputStream`, so it has to be closed explicitly.

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://issues.liferay.com/browse/LPS-182336

Thank you and best regards,
István

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/RenderedImageUtil.java [hunk 1]
```diff
@@ -105,6 +105,14 @@ public static RenderedImage readImage(InputStream inputStream)
...
 					imageReader.dispose();
 				}
+
+				if (imageInputStream != null) {
+					imageInputStream.close();
+				}
+
+				if (inputStream != null) {
+					inputStream.close();
+				}
```
Comment (adolfopa) [inline modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/RenderedImageUtil.java:115]: Can you use StreamUtil.cleanUp here?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/RenderedImageUtil.java (hunk 1 above)
Comment (IstvanD) [reply modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/RenderedImageUtil.java]: @adolfopa I force-pushed it.

**Comment by Hanlf** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**